cmd/lakectl/cmd/commit.go

```diff
@@ -9,31 +9,43 @@ import (
 	"github.com/treeverse/lakefs/pkg/uri"
 )
 
-var commitCreateTemplate = `Commit for branch "{{.Branch.Ref}}" completed.
+var (
+	errInvalidKeyValueFormat    = fmt.Errorf("invalid key/value pair - should be separated by \"=\"")
+	errEmptyMessage             = fmt.Errorf("commit with an empty message without specifying the \"--allow-empty-message\" flag")
+	allowEmptyCommitMessageFlag bool
+	commitCreateTemplate        = `Commit for branch "{{.Branch.Ref}}" completed.
 
 ID: {{.Commit.Id|yellow}}
 Message: {{.Commit.Message}}
 Timestamp: {{.Commit.CreationDate|date}}
 Parents: {{.Commit.Parents|join ", "}}
 
 `
+)
 
-var errInvalidKeyValueFormat = fmt.Errorf("invalid key/value pair - should be separated by \"=\"")
+const (
+	MessageFlagName           string = "message"
+	AllowEmptyMessageFlagName string = "allow-empty-message"
+	MetaFlagName              string = "meta"
+)
 
 var commitCmd = &cobra.Command{
 	Use:   "commit <branch uri>",
 	Short: "Commit changes on a given branch",
 	Args:  cobra.ExactArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
 		// validate message
-		kvPairs, err := getKV(cmd, "meta")
+		kvPairs, err := getKV(cmd, MetaFlagName)
 		if err != nil {
 			DieErr(err)
 		}
-		message, err := cmd.Flags().GetString("message")
+		message, err := cmd.Flags().GetString(MessageFlagName)
 		if err != nil {
 			DieErr(err)
 		}
+		if strings.TrimSpace(message) == "" && !allowEmptyCommitMessageFlag {
```

Comment:
prefer to check the boolean before you do any work - just if eval order and work